### ISSUE TITLE
Build Rust CLI scorer application

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1967]
## Summary

Create a native Rust CLI binary that scores a creature against a training data directory, replicating the full scoring pipeline currently implemented in TypeScript.

## Context

The current scoring pipeline runs in TypeScript: load creature JSON → compile to WASM network → iterate training records → activate network → calculate loss → compute final score with complexity penalty. This issue creates a pure Rust equivalent as a standalone binary, eliminating all JS/WASM boundary overhead.

## What needs to be done

1. Create a new binary crate in the Cargo workspace (e.g., `rust_scorer/`)
2. Implement CLI argument parsing (using `clap`) with parameters:
   - `--creature <path>`: Path to creature JSON file
   - `--data <path>`: Path to training data directory
   - `--cost <name>`: Cost function name (MSE, MAE, CrossEntropy, MAPE, MSLE, HINGE)
   - `--inputs <n>`: Number of input neurons
   - `--outputs <n>`: Number of output neurons
   - `--growth-cost <f>`: Growth cost factor for complexity penalty (default: 0.0001)
3. Implement the scoring pipeline:
   a. Parse creature JSON → build `CompiledNetwork`
   b. Read all training records from the data directory
   c. For each record: activate network, calculate loss against expected output
   d. Compute average error across all records
   e. Calculate complexity penalty:
      - `hiddenNeuronCount × growthCost`
      - `synapseCount × growthCost / 10`
      - `(valuePenalty(maxWeightBias) + valuePenalty(avgWeightBias)) / 2 × growthCost / 100`
      - `squashComplexityPenalty × growthCost / 100`
   f. Calculate version penalty (if semantic version mismatch)
   g. Final score = `1 - error - complexityPenalty - versionPenalty`
4. Output result as JSON to stdout:
   ```json
   {
     "score": 0.85,
     "error": 0.12,
     "complexityPenalty": 0.03,
     "recordCount": 5000,
     "hiddenNeurons": 150,
     "synapseCount": 2000
   }
   ```
5. Implement the `valuePenalty` function matching the TS implementation:
   - `if value <= 1: return 0`
   - `primaryPenalty = 1 / (1 + 1/value)`
   - Recursive compression for values > 0.999

## Acceptance criteria

- [ ] Compiles as a native binary for the host platform
- [ ] Accepts command-line arguments for creature path, data directory, cost function, and dimensions
- [ ] Produces a JSON score output on stdout
- [ ] Implements all 6 cost functions (MSE, MAE, CrossEntropy, MAPE, MSLE, HINGE)
- [ ] Score calculation matches the TS formula exactly (within floating-point tolerance)
- [ ] Exits with non-zero status on invalid input (missing files, dimension mismatch, etc.)
- [ ] Use Australian English in all code and comments

Depends on #1965
Depends on #1966

Part of #1963

**Comment by stservice** [comment]
Claimed by `GRQ-25`

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/1982 has been created to address this issue.

**Comment by stservice** [comment]
Automatically closed — PR #1982 has been merged.
